src/txdb.cpp

```diff
@@ -321,7 +320,9 @@ bool CBlockTreeDB::LoadBlockIndexGuts(const Consensus::Params& consensusParams,
                 pindexNew->nTx            = diskindex.nTx;
 
                 const uint256 block_hash = pindexNew->GetBlockHash();
-                if (!CheckProof(pindexNew->GetBlockHeader(), consensusParams) &&
+                // Only validate one of every 1000 block header for sanity check
+                if (pindexNew->nHeight % 1000 == 0 &&
+                        !CheckProof(pindexNew->GetBlockHeader(), consensusParams) &&
```

Comment:
I don't observe a significant slowdown on 0.14.1 as I did on 0.17. But yeah it can only improve things, I guess.